examples/tsc/BUILD.gn

```diff
@@ -0,0 +1,201 @@
+  "$rust_build:tokio_executor",
+  "$rust_build:tokio_fs",
+  "$rust_build:tokio_io",
+  "$rust_build:tokio_process",
+  "$rust_build:tokio_threadpool",
+  "$rust_build:url",
+]
+if (is_win) {
+  main_extern += [ "$rust_build:winapi" ]
+}
+
+ts_sources = [
+  "main.ts",
+  "../../js/assets.ts",
+  "../../js/blob.ts",
+  "../../js/buffer.ts",
+  "../../js/chmod.ts",
+  "../../js/console_table.ts",
+  "../../js/compiler.ts",
+  "../../js/console.ts",
+  "../../js/copy_file.ts",
+  "../../js/custom_event.ts",
+  "../../js/deno.ts",
+  "../../js/dir.ts",
+  "../../js/dispatch.ts",
+  "../../js/dom_types.ts",
+  "../../js/errors.ts",
+  "../../js/event.ts",
+  "../../js/event_target.ts",
+  "../../js/fetch.ts",
+  "../../js/format_error.ts",
+  "../../js/dom_file.ts",
+  "../../js/file_info.ts",
+  "../../js/files.ts",
+  "../../js/flatbuffers.ts",
+  "../../js/form_data.ts",
+  "../../js/globals.ts",
+  "../../js/headers.ts",
+  "../../js/io.ts",
+  "../../js/libdeno.ts",
+  "../../js/lib.web_assembly.d.ts",
+  "../../js/location.ts",
+  "../../js/main.ts",
+  "../../js/make_temp_dir.ts",
+  "../../js/metrics.ts",
+  "../../js/mkdir.ts",
+  "../../js/mock_builtin.js",
+  "../../js/net.ts",
+  "../../js/os.ts",
+  "../../js/platform.ts",
+  "../../js/plugins.d.ts",
+  "../../js/process.ts",
+  "../../js/read_dir.ts",
+  "../../js/read_file.ts",
+  "../../js/read_link.ts",
+  "../../js/remove.ts",
+  "../../js/rename.ts",
+  "../../js/repl.ts",
+  "../../js/resources.ts",
+  "../../js/stat.ts",
+  "../../js/symlink.ts",
+  "../../js/text_encoding.ts",
+  "../../js/timers.ts",
+  "../../js/truncate.ts",
+  "../../js/types.ts",
+  "../../js/url.ts",
+  "../../js/url_search_params.ts",
+  "../../js/util.ts",
+  "../../js/workers.ts",
+  "../../js/write_file.ts",
+  "../../js/performance.ts",
+  "../../js/version.ts",
+  "../../tsconfig.json",
+
+  # Listing package.json and yarn.lock as sources ensures the bundle is rebuilt
+  # when npm packages are added/removed or their contents changes.
+  "../../package.json",
+  "../../third_party/yarn.lock",
+]
+
+# When Cargo is driving the build, GN/Ninja are used to produce these non-Rust
+# targets. Cargo handles all Rust source files and the final linking step.
+group("tsc_deps") {
+  deps = [
+    ":msg_rs",
+    ":snapshot_tsc",
+    "../../libdeno:libdeno_static_lib",
+  ]
+}
+
+# Reads the cargo info from Cargo.toml
+deno_cargo_info = exec_script("//build_extra/rust/get_cargo_info.py",
+                              [ rebase_path("../../Cargo.toml", root_build_dir) ],
+                              "json")
+
+rust_executable("tsc") {
+  source_root = "main.rs"
+  extern = main_extern
+  deps = [
+    ":tsc_deps",
+  ]
+
+  crate_version = deno_cargo_info.version
+}
+
+rust_test("test_tsc") {
+  source_root = "main.rs"
+  extern = main_extern
+  deps = [
+    ":tsc_deps",
+  ]
+
+  # Extract version from Cargo.toml
+  crate_version = deno_cargo_info.version
+}
+
+run_node("tsc_bundle") {
```

Comment:
We should really have a `template("bundle")`...

Reply:
It is too opinionated about outputs, I tried. I don't know how easy it would be to update the template, but I will try.

Reply:
Ok - not high priority.